src/date-range-picker/date-range-picker-embedded.tsx

```diff
@@ -0,0 +1,149 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import { DateRangePickerProps } from './interfaces';
+import Calendar, { DayIndex } from './calendar';
+import InternalBox from '../box/internal';
+import SpaceBetween from '../space-between/index.js';
+
+import styles from './styles.css.js';
+import RelativeRangePicker from './relative-range';
+import ModeSwitcher from './mode-switcher';
+import clsx from 'clsx';
+import InternalAlert from '../alert/internal';
+import LiveRegion from '../internal/components/live-region';
+import { useDateRangePicker } from './use-date-range-picker';
+
+export interface DateRangePickerDropdownProps
+  extends Pick<
+    Required<DateRangePickerProps>,
+    | 'locale'
+    | 'isDateEnabled'
+    | 'isValidRange'
+    | 'value'
+    | 'relativeOptions'
+    | 'i18nStrings'
+    | 'dateOnly'
+    | 'timeInputFormat'
+    | 'rangeSelectorMode'
+  > {
+  startOfWeek: DayIndex;
+  isSingleGrid: boolean;
+
+  ariaLabelledby?: string;
+  ariaDescribedby?: string;
+}
+
+export function DateRangePickerEmbedded({
+  locale = '',
+  startOfWeek,
+  isDateEnabled = () => true,
+  isValidRange = () => ({ valid: true }),
+  value,
+  relativeOptions,
+  isSingleGrid,
+  i18nStrings,
+  dateOnly,
+  timeInputFormat,
+  rangeSelectorMode,
+  ariaLabelledby,
+  ariaDescribedby,
+}: DateRangePickerDropdownProps) {
+  const {
+    focusVisible,
+    focusRefs,
+    scrollableContainerRef,
+    rangeSelectionMode,
+    setRangeSelectionMode,
+    selectedAbsoluteRange,
+    setSelectedAbsoluteRange,
+    selectedRelativeRange,
+    setSelectedRelativeRange,
+    validationResult,
+    setValidationResult,
+    VALID_RANGE,
+  } = useDateRangePicker({
+    value,
+    relativeOptions,
+    rangeSelectorMode,
+    isValidRange,
+  });
+
+  return (
+    <div
```

Comment:
Looks like a lot of the markup is duplicated between date-range-picker-embedded and the dropdown. It would be better if the dropdown can be shared instead:

```
// date-range-picker-embedded.tsx

function DateRangePickerEmbedded(props) {
  const drpProps = useDateRangePicker(props)
  ...
  return <Dropdown ... />
}

// internal.tsx

function DateRangePicker(props) {
  const drpProps = useDateRangePicker(props)
  ...
  return <>
    <Trigger ... />
    <Dropdown ... />
  <>
}
```